src/Microsoft.ML.PCA/PcaTransform.cs

```diff
@@ -190,90 +281,104 @@ private static VersionInfo GetVersionInfo()
         {
             return new VersionInfo(
                 modelSignature: "PCA FUNC",
-                verWrittenCur: 0x00010001, // Initial
+                //verWrittenCur: 0x00010001, // Initial
+                verWrittenCur: 0x00010002, // Get rid of writing float size in model context
                 verReadableCur: 0x00010001,
                 verWeCanReadBack: 0x00010001,
                 loaderSignature: LoaderSignature,
                 loaderAssemblyName: typeof(PcaTransform).Assembly.FullName);
         }
 
-        // These are parallel to Infos.
-        private readonly ColumnType[] _types;
+        private readonly int _numColumns;
+        private readonly ColumnInfo[] _columns;
         private readonly TransformInfo[] _transformInfos;
 
-        private readonly int[] _oversampling;
-        private readonly bool[] _center;
-        private readonly int[] _weightColumnIndex;
-
         private const string RegistrationName = "Pca";
 
         /// <summary>
         /// Public constructor corresponding to SignatureDataTransform.
         /// </summary>
-        public PcaTransform(IHostEnvironment env, Arguments args, IDataView input)
-            : base(env, RegistrationName, Contracts.CheckRef(args, nameof(args)).Column,
-                input, TestIsFloatItem)
+        public PcaTransform(IHostEnvironment env, IDataView input, ColumnInfo[] columns)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(PcaTransform)), GetColumnPairs(columns))
         {
-            Host.AssertNonEmpty(Infos);
-            Host.Assert(Infos.Length == Utils.Size(args.Column));
+            Host.AssertNonEmpty(ColumnPairs);
 
-            _transformInfos = new TransformInfo[args.Column.Length];
-            _oversampling = new int[args.Column.Length];
-            _center = new bool[args.Column.Length];
-            _weightColumnIndex = new int[args.Column.Length];
-            for (int i = 0; i < _transformInfos.Length; i++)
+            _columns = columns;
+            _numColumns = columns.Length;
+            _transformInfos = new TransformInfo[_numColumns];
+
+            for (int i = 0; i < _numColumns; i++)
             {
-                Host.Check(Infos[i].TypeSrc.VectorSize > 1, "Pca transform can only be applied to columns with known dimensionality greater than 1");
-                _transformInfos[i] = new TransformInfo(args.Column[i], args, Infos[i].TypeSrc.ValueCount);
-                _center[i] = args.Column[i].Center ?? args.Center;
-                _oversampling[i] = args.Column[i].Oversampling ?? args.Oversampling;
-                Host.CheckUserArg(_oversampling[i] >= 0, nameof(args.Oversampling), "Oversampling must be non-negative");
```

Comment:
>Oversampling must be non-negative" [](start = 85, length = 34)

can you put this check inside ColumnInfo constructor?

Reply:
Moved to ctor

---
In reply to: [227051899](https://github.com/dotnet/machinelearning/pull/1333#discussion_r227051899) [](ancestors = 227051899)